### DISCUSSION TITLE
Improve DX during the setting update

### DISCUSSION THREAD

**Discussion by qdequele** [meilisearch/product#559] [Feedback & Feature Proposal]
For a demo, I have to do on the Hacker News data I had an unexpected experience. After indexing all the 32M documents, I understood that I had to add one `filterableAttributes` and two `displayedAttributes`. Unfortunately to see the change on the `displayedAttributes` I had to way a few hours before the changes on `filterableAttributes` finish. 

Do you thing it would be possible to do the settings change in multiple times, starting by what doesn't need a reindexation?

**Comment by irevoire** [comment]
Unfortunately, I don’t think there is a good way to do this because if we split your settings and apply one part before the other and the other fail. Then we end up in an inconsistent state, and I don’t even know what to return to the user on the `tasks` endpoint 🤔
